conformance/tests/gateway-invalid-tls-certificateref.go

```diff
@@ -47,6 +47,7 @@ var GatewayInvalidTLSConfiguration = suite.ConformanceTest{
 				Status: metav1.ConditionFalse,
 				Reason: string(v1beta1.ListenerReasonInvalidCertificateRef),
 			}},
+			AttachedRoutes: 1,
```

Comment:
I actually would expect `AttachedRoutes` to be 0 in this case. Did you happen to test this on any implementation? If so, are there any that are currently setting the value to 1 here? If a TLS Cert is invalid, I would not expect Routes to be considered attached. Maybe there would be an exception in the case of a Gateway with a cert that was previously valid, but I'm not seeing that here.

Reply:
Thanks @robscott for the feedback! I was testing this on istio implementation. I am not very sure if there is anything set the value to 1. I may need to look into the setting of istio implementation.

Reply:
> If a TLS Cert is invalid, I would not expect Routes to be considered attached.

@robscott I agree it should be 0 here, since there are no routes attached in this test case, but in general shouldn't the `attachedRoutes` be based on how many routes are connected to the gateway, independent of whether or not the gateway itself is currently ready/valid?

Reply:
The current spec for `attachedRoutes` is pretty light here, but it does require the Routes to have been "successfully attached to this listener":

https://github.com/kubernetes-sigs/gateway-api/blob/46d40889ea70291da0614274931c2ffacdc0b612/apis/v1beta1/gateway_types.go#L671-L673

In this case I can think there's some ambiguity in terms of what happens when the Listener itself is invalid but nothing is invalid with the Listener->Route relationship. This can be worked out in a separate PR, but I've been thinking this is how we should interpret "AttachedRoutes:

>AttachedRoutes represents the total number of Routes that have been successfully attached to this Listener. For the purposes of this condition, Routes MUST only be considered "Attached" if the Listener has both the "Accepted" and "ResolvedRefs" conditions set to "True" AND the Route has the "Accepted" condition set to "True". If one or more  of these Listener conditions transitions from "True" to "False", this field SHOULD represent the number of Routes that have been attached to the Listener before this change. Routes that are intentionally detached from a Gateway (either through changes to AllowedRoutes on the Listener or ParentRefs on the Routes) MUST NOT be counted.

@howardjohn do you know how Istio is interpreting this today? At first glance, it looks like it's considering any Route that would be attached, regardless of the validity of the Listener itself.

Reply:
IMO, the Listener<->Route attachment is valid even if the listener is not valid - or at least that was my original thought. The revised description sounds good from a user perspective but its not implementable (for us, anyways) because it requires looking back in history instead of reconciling the current state of the world

Reply:
@robscott It seems simpler to me to keep them independent. Otherwise, it's confusing when a route status says the route is "Accepted", but it's not acknowledged as attached in the gateway. Maybe the route should not be "Accepted" if the listener is not ready, but I think the doc would then need to explicitly say that. The current doc seems to imply that a route could be accepted/attached to a gateway, but still not "available" depending on the gateway/listener status, which seems like a clean and simple approach.
https://github.com/kubernetes-sigs/gateway-api/blob/46d40889ea70291da0614274931c2ffacdc0b612/apis/v1beta1/shared_types.go#L315-L316

Reply:
x-ref https://github.com/kubernetes-sigs/gateway-api/issues/1626 for further discussion.

Reply:
Thanks for the discussion! Updated to 0 here. Sorry for the late response. I was busy taking care of my newborn in the past month.